packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart

```diff
@@ -122,20 +122,36 @@ abstract class RenderSliverBoxChildManager {
   bool debugAssertChildListLocked() => true;
 }
 
+/// Parent data structure used by [RenderSliverMultiKeepAliveBoxAdaptor].
+class SliverMultiKeepAliveBoxAdaptorParentData extends SliverLogicalParentData  {
+  /// Whether to keep the child alive even when it is no longer visible.
+  bool keepAlive = false;
+
+  @override
+  String toString() => '${keepAlive == true ? "keepAlive; " : ""}${super.toString()}';
+}
+
 /// Parent data structure used by [RenderSliverMultiBoxAdaptor].
-class SliverMultiBoxAdaptorParentData extends SliverLogicalParentData with ContainerParentDataMixin<RenderBox> {
+class SliverMultiBoxAdaptorParentData extends SliverMultiKeepAliveBoxAdaptorParentData with ContainerParentDataMixin<RenderBox> {
   /// The index of this child according to the [RenderSliverBoxChildManager].
   int index;
 
-  /// Whether to keep the child alive even when it is no longer visible.
-  bool keepAlive = false;
-
   /// Whether the widget is currently in the
   /// [RenderSliverMultiBoxAdaptor._keepAliveBucket].
   bool _keptAlive = false;
 
   @override
-  String toString() => 'index=$index; ${keepAlive == true ? "keepAlive; " : ""}${super.toString()}';
+  String toString() => 'index=$index; ${super.toString()}';
+}
+
+/// A sliver with multiple box children that can be kept alive.
+abstract class RenderSliverMultiKeepAliveBoxAdaptor extends RenderSliver{
```

Comment:
`RenderSliverMultiBoxAdaptorWithKeepAlive` would be easier to understand, i think.

But it's not clear why this one has to be Multi anything. It doesn't introduce that opinion. Maybe just "RenderSliverWithKeepAlive"?